components/control/bluetooth-buttons/README.md

```diff
@@ -0,0 +1,82 @@
+	- If later changing the headset, re-run `$ ./register-device.py`. Reboot or restart the service with `sudo systemctl restart phoniebox-bt-buttons.service `
+
+### Supported Buttons
+
+Out-of-the box support is included for the following buttons
+
+- Play/Pause
+- Previous Track
+- Next Track
+
+Key codes are standarized and so it should also work with your headphones. If you want to add more keys or assign a different behaviour see [Troubleshooting](#troubleshooting)
+
+*Note:* Volume up/down is inherently supported by the bluetooth protocol. There is no need to handle these by this script.
+
+### On Connect / On Disconnect
+
+If the feature [bluetooth-sink-switch](../../bluetooth-sink-switch) is enabled, the script automatically switches the audio stream to headphones / regular speakers on bluetooth connect / disconnect  respectivly. Playback state (play/pause) is retained.
+
+*Note:* On-connect actions may take up to 4 seconds - please be patient (bluetooth connection is only checked every two seconds, bluetooth stream needs to be buffered, etc...)
+
+You can **customize** the behaviour by editing the functions 
+
+- `bt_on_connect(mpd_support=0)`
+- `bt_on_disconnect(mpd_support=0)`
+
+where `mpd_support` indicates wether the bt-sink-switch-feature is enabled.
+
+### Troubleshooting
+
+This feature has been tested with PowerLocus Buddy and Sennheiser Momentum M2 AEBT headphones.
+
+#### Preparation
+
+- Stop the service `$ sudo systemctl stop phoniebox-bt-buttons.service`
+- Start the script in a command line with debug option `$ ./bt-buttons.py debug`
+
+#### Check that correct bluetooth device is found
+
+- Run the [preparatory steps](#preparation)
+- Check headset is connected and listed as input event device with `$ cat /proc/bus/input/devices`. Note the device name.
+- In the script's debug output you should see something like this. Here the MAC address is the device name
+~~~
+30.12.2020 21:44:41 - bt-buttons.py - DEBUG: bt_get_device_name() -> C4:FB:20:63:A7:F2
+30.12.2020 21:45:05 - bt-buttons.py - DEBUG: bt_open_device(C4:FB:20:63:A7:F2): Device 'C4:FB:20:63:A7:F2' search success
+30.12.2020 21:45:05 - bt-buttons.py - DEBUG: device /dev/input/event1, name "C4:FB:20:63:A7:F2", phys ""
+~~~
+
+- If you see discrepancies, re-run `$ ./register-device.py`(see above)
+
+#### Add key codes / change actions
+- Run the [preparatory steps](#preparation)
+- Press the buttons on the headset and check for these debug outputs. Note down the keycode. The **163** is the keycode, you are looking for. Go through all the buttons. Also try short/long press. On my headphones, they result in different keycodes
+```
+30.12.2020 21:45:59 - bt-buttons.py - DEBUG: key event at 1609361159.529679, 163 (KEY_NEXTSONG), down
+```
+- Go into the source code and adjust these lines for desired behaviour
+~~~python
+                if event.code == bt_keycode_play:
```

Comment:
Would it make sense to have a config file? Editing source files has the disadvantages that inexperienced users may accidentally break the code and that the git workspace now contains modified files.

Reply:
Firstly, key codes a standardized and existing key codes should work with any headset. I have mapped all key codes of the Bluetooth headphones I could lay my fingers in the script. If somebody has more, they could add them and make a pull request. The script should work unmodified for the vast majority of use cases.  

Cross-mapping buttons to a different behaviour is of course possible, but not the main use case. 

Inexperienced uses get a chance to learn some Python :-) Having a modified script file is of course not ideal, but should no happen unless you do a cross-mapping. 

A generic config file is a lot of work to set up. I see this as possible future enhancement.

Reply:
Agreed, leaving that as future improvement.